### ISSUE TITLE
Using custom emoji with stateful variable does not update properly

### ISSUE THREAD

**Issue by ePirat** [missive/emoji-mart#556]
I am loading the custom emoji list from a server into a stateful variable, like this:

```jsx
function EmojiPicker() {
    let [emoji, setEmoji] = React.useState(null);

    React.useEffect(async () => {
        let customEmoji = await getEmoji();
        setEmoji(customEmoji);
    }, []);

    return (
        <Picker native={true} custom={emoji} />
    );
}
```

However this does not update the custom emoji shown in the picker. While for the initial load I can just delay mounting the picker, I can not figure out how to handle update of the list.

Is this not supported or am I doing something wrong here?

**Comment by TranquilMarmot** [comment]
I ran into this as well. The only way I could get around it was to not render the component until the custom emoji are ready.

In your example, you could try...

```diff
function EmojiPicker() {
    let [emoji, setEmoji] = React.useState(null);

    React.useEffect(async () => {
        let customEmoji = await getEmoji();
        setEmoji(customEmoji);
    }, []);

    return (
-        <Picker native={true} custom={emoji} />
+        emoji && <Picker native={true} custom={emoji} />
    );
}
```

**Comment by test137E29B** [comment]
Something along the lines of 
```js
const PickerComponent = React.useMemo(() => <Picker native custom={emoji} />, [emoji]);

return <PickerComponent />;
```
might work better, as it'll always show the component, custom loaded or otherwise. Not tested this though, so unsure of behaviors when the emoji do get set, and the memo updates.

**Comment by EtienneLem** [comment]
React issues have been addressed in **v5.2.0**. You can use [`@emoji-mart/react`](https://github.com/missive/emoji-mart#react) now
